### DISCUSSION TITLE
Improve: Insert a suggestion

### DISCUSSION THREAD

**Discussion by tupui** [community/community#4524] [Other Feature Feedback, Questions, & Ideas]
I like the ability when reviewing to directly suggest the changes I would like to have.

There are some things that might be improved. From must have to nice to have.

* Spacing: I am not sure what is happening here, but when doing multi-line changes, it is very difficult to have the correct spacing. I must click preview all the time and do this back and forth while adding a few spaces.
* Allow edit of more code, not just around the current changes. I often need to add an `import` for instance.
* If someone did not change the default commit message when applying suggestions, and apply a bunch of them in a row, merge them automatically.
* Add live-preview.
* Add code completion.

Basically, the ideal solution to me would almost look like opening an IDE 😅

In general this feature is great and I really like that it adds co-author information!
